core/codenvy-hosted-api-permission/src/main/java/com/codenvy/api/permission/server/PermissionManager.java

```diff
@@ -0,0 +1,158 @@
+/*
+ *  [2012] - [2016] Codenvy, S.A.
+ *  All Rights Reserved.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of Codenvy S.A. and its suppliers,
+ * if any.  The intellectual and technical concepts contained
+ * herein are proprietary to Codenvy S.A.
+ * and its suppliers and may be covered by U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+package com.codenvy.api.permission.server;
+
+import com.codenvy.api.permission.server.dao.PermissionsStorage;
+
+import org.eclipse.che.api.core.ConflictException;
+import org.eclipse.che.api.core.ServerException;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Class that holds supported {@link PermissionsStorage} and delegates calls to them
+ *
+ * @author gazarenkov
+ * @author Sergii Leschenko
+ */
+@Singleton
+public class PermissionManager {
+    private final Map<String, PermissionsStorage> domainToStorage = new HashMap<>();
+    private final Map<String, PermissionsDomain>  domains         = new HashMap<>();
+
+    @Inject
+    public PermissionManager(Set<PermissionsStorage> storages) throws ServerException {
+        for (PermissionsStorage storage : storages) {
+            for (PermissionsDomain domain : storage.getDomains()) {
+                domains.put(domain.getId(), domain);
+                PermissionsStorage oldStorage = domainToStorage.put(domain.getId(), storage);
+                if (oldStorage != null) {
+                    throw new ServerException("Permissions Domain '" + domain.getId() + "' should be stored in only one storage. " +
+                                              "Duplicated in " + storage.getClass() + " and " + oldStorage.getClass());
+                }
+            }
+        }
+    }
+
+    /**
+     * @see PermissionsStorage#store(Permissions)
```

Comment:
Why clients of that class should read docs for low level operations with permissions?